### PR TITLE
ui: Add filter compaction UI

### PR DESCRIPTION
**Pull request by adzshaf**
* [x] I added CHANGELOG entry for this change.
* [ ] Change is not relevant to the end user.

## Changes

This PR is part of LFX Mentorship project to improve Block Viewer UI (issue: #3112). We want to add feature so user can filter blocks by compaction level.

## Verification

When we haven't filter anything yet:
image

Filter by compaction level:
image

Tested on local using `make quickstart`

cc: @onprem @squat

### CODE REVIEW

**Comment by yeya24** [comment]
Looks nice! But why the page title is `Thanos - Query`?

**Comment by onprem** [comment]
> Looks nice! But why the page title is `Thanos - Query`?

This is a side effect of how we choose to do development locally. The UI in dev mode just assumes that it is running for Querier.

**Comment by yeya24** [comment]
This looks good. @adzshaf Can you please resolve the conflict?

**Comment by yeya24** [comment]
Any other work needs to be done for this pr? @onprem If not I am happy to resolve the conflicts and get this feature merged.

File: pkg/ui/react-app/src/thanos/pages/blocks/SourceView.tsx [hunk 1]
```diff
@@ -2,16 +2,18 @@ import React, { FC } from 'react';
...
   blocks: Block[];
   gridMinTime: number;
   gridMaxTime: number;
   selectBlock: React.Dispatch<React.SetStateAction<Block | undefined>>;
   blockSearch: string;
-}> = ({ blocks, gridMinTime, gridMaxTime, selectBlock, blockSearch }) => {
-  const blockSearchValue = getBlockByUlid(blocks, blockSearch);
+  compactionLevel: number;
+}> = ({ blocks, gridMinTime, gridMaxTime, selectBlock, blockSearch, compactionLevel }) => {
+  let blockSearchValue = getBlockByUlid(blocks, blockSearch);
```
Comment (onprem) [inline pkg/ui/react-app/src/thanos/pages/blocks/SourceView.tsx:15]:
```suggestion
  let filteredBlocks = getBlockByUlid(blocks, blockSearch);
```

**Review by onprem — APPROVED** [review]
LGTM! Just a non-blocking suggestion. This only needs a rebase and then we can merge this.

File: pkg/ui/react-app/src/thanos/pages/blocks/SourceView.tsx (hunk 1 above)
Comment (yeya24) [reply pkg/ui/react-app/src/thanos/pages/blocks/SourceView.tsx]: Fixed in the latest commit

**Review by yeya24 — APPROVED** [review]
Docs CI failure not related. Merging now…

**Pull request merged**